From f05af2de459d95e27d5605db7969ba692b3de918 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sun, 19 Apr 2020 04:13:51 +0300
Subject: [PATCH] ThemePicker: Showcase the Font name

  * from
  https://github.com/crdroidandroid/android_packages_apps_ThemePicker/commit/acd2609b4386e428fc6f146ed229036cc7854a2d

	modified:   res/layout/preview_card_font_content.xml
	modified:   src/com/android/customization/model/theme/OverlayThemeExtractor.java
	modified:   src/com/android/customization/model/theme/ThemeBundle.java
	modified:   src/com/android/customization/model/theme/custom/ThemeComponentOption.java
	modified:   src/com/android/customization/picker/theme/ThemeFragment.java

Change-Id: I1d6d3a46651c370c983bd304f33cf58c0ea3a731
---
 res/layout/preview_card_font_content.xml           | 10 +++++++++-
 .../model/theme/OverlayThemeExtractor.java         | 10 +++++++---
 .../customization/model/theme/ThemeBundle.java     | 14 +++++++++++---
 .../model/theme/custom/ThemeComponentOption.java   |  6 +++++-
 .../customization/picker/theme/ThemeFragment.java  |  3 +++
 5 files changed, 35 insertions(+), 8 deletions(-)

diff --git a/res/layout/preview_card_font_content.xml b/res/layout/preview_card_font_content.xml
index fb7879e..69138e9 100644
--- a/res/layout/preview_card_font_content.xml
+++ b/res/layout/preview_card_font_content.xml
@@ -55,4 +55,12 @@
         android:layout_gravity="bottom|center_horizontal"
         android:gravity="center_horizontal"
         android:text="@string/font_card_body"/>
-</LinearLayout>
\ No newline at end of file
+    <TextView
+        style="@style/FontCardBodyTextStyle"
+        android:id="@+id/font_card_body_name"
+        android:layout_height="56sp"
+        android:layout_weight="2"
+        android:layout_gravity="bottom|center_horizontal"
+        android:gravity="center_horizontal"
+        android:text=""/>
+</LinearLayout>
diff --git a/src/com/android/customization/model/theme/OverlayThemeExtractor.java b/src/com/android/customization/model/theme/OverlayThemeExtractor.java
index 9c85fa4..bb1cba8 100644
--- a/src/com/android/customization/model/theme/OverlayThemeExtractor.java
+++ b/src/com/android/customization/model/theme/OverlayThemeExtractor.java
@@ -150,7 +150,10 @@ class OverlayThemeExtractor {
                             fontOverlayPackage))
                     .setHeadlineFontFamily(loadTypeface(
                             ResourceConstants.CONFIG_HEADLINE_FONT_FAMILY,
-                            fontOverlayPackage));
+                            fontOverlayPackage))
+                    .setFontName(mContext.getPackageManager()
+                        .getPackageInfo(fontOverlayPackage, 0)
+                        .applicationInfo.loadLabel(mContext.getPackageManager()).toString());
         } else {
             addSystemDefaultFont(builder);
         }
@@ -201,7 +204,8 @@ class OverlayThemeExtractor {
                 ResourceConstants.CONFIG_BODY_FONT_FAMILY,
                 "string", ResourceConstants.ANDROID_PACKAGE));
         builder.setHeadlineFontFamily(Typeface.create(headlineFontFamily, Typeface.NORMAL))
-                .setBodyFontFamily(Typeface.create(bodyFontFamily, Typeface.NORMAL));
+                .setBodyFontFamily(Typeface.create(bodyFontFamily, Typeface.NORMAL))
+                .setFontName(mContext.getString(R.string.default_theme_title));
     }
 
     Typeface loadTypeface(String configName, String fontOverlayPackage)
@@ -276,4 +280,4 @@ class OverlayThemeExtractor {
     String[] getShapePreviewIconPackages() {
         return mShapePreviewIconPackages;
     }
-}
\ No newline at end of file
+}
diff --git a/src/com/android/customization/model/theme/ThemeBundle.java b/src/com/android/customization/model/theme/ThemeBundle.java
index e83e852..a5e41df 100644
--- a/src/com/android/customization/model/theme/ThemeBundle.java
+++ b/src/com/android/customization/model/theme/ThemeBundle.java
@@ -276,6 +276,7 @@ public class ThemeBundle implements CustomizationOption<ThemeBundle> {
     public static class PreviewInfo {
         public final Typeface bodyFontFamily;
         public final Typeface headlineFontFamily;
+        public final String fontName;
         @ColorInt public final int colorAccentLight;
         @ColorInt public final int colorAccentDark;
         public final List<Drawable> icons;
@@ -285,11 +286,12 @@ public class ThemeBundle implements CustomizationOption<ThemeBundle> {
         @Dimension public final int bottomSheeetCornerRadius;
 
         private PreviewInfo(Context context, Typeface bodyFontFamily, Typeface headlineFontFamily,
-                int colorAccentLight, int colorAccentDark, List<Drawable> icons,
+                String fontName, int colorAccentLight, int colorAccentDark, List<Drawable> icons,
                 Drawable shapeDrawable, @Dimension int cornerRadius,
                 @Nullable Asset wallpaperAsset, List<Drawable> shapeAppIcons) {
             this.bodyFontFamily = bodyFontFamily;
             this.headlineFontFamily = headlineFontFamily;
+            this.fontName = fontName;
             this.colorAccentLight = colorAccentLight;
             this.colorAccentDark = colorAccentDark;
             this.icons = icons;
@@ -316,6 +318,7 @@ public class ThemeBundle implements CustomizationOption<ThemeBundle> {
         protected String mTitle;
         private Typeface mBodyFontFamily;
         private Typeface mHeadlineFontFamily;
+        private String mFontName;
         @ColorInt private int mColorAccentLight = -1;
         @ColorInt private int mColorAccentDark = -1;
         private List<Drawable> mIcons = new ArrayList<>();
@@ -358,8 +361,8 @@ public class ThemeBundle implements CustomizationOption<ThemeBundle> {
                     //  non-adaptive icons
                 }
             }
-            return new PreviewInfo(context, mBodyFontFamily, mHeadlineFontFamily, mColorAccentLight,
-                    mColorAccentDark, mIcons, shapeDrawable, mCornerRadius,
+            return new PreviewInfo(context, mBodyFontFamily, mHeadlineFontFamily, mFontName,
+                    mColorAccentLight, mColorAccentDark, mIcons, shapeDrawable, mCornerRadius,
                     mWallpaperAsset, shapeIcons);
         }
 
@@ -386,6 +389,11 @@ public class ThemeBundle implements CustomizationOption<ThemeBundle> {
             return this;
         }
 
+        public Builder setFontName(@Nullable String fontName) {
+            mFontName = fontName;
+            return this;
+        }
+
         public Builder setColorAccentLight(@ColorInt int colorAccentLight) {
             mColorAccentLight = colorAccentLight;
             return this;
diff --git a/src/com/android/customization/model/theme/custom/ThemeComponentOption.java b/src/com/android/customization/model/theme/custom/ThemeComponentOption.java
index 6583433..910563c 100644
--- a/src/com/android/customization/model/theme/custom/ThemeComponentOption.java
+++ b/src/com/android/customization/model/theme/custom/ThemeComponentOption.java
@@ -148,13 +148,17 @@ public abstract class ThemeComponentOption implements CustomizationOption<ThemeC
             title.setTypeface(mHeadlineFont);
             TextView bodyText = container.findViewById(R.id.font_card_body);
             bodyText.setTypeface(mBodyFont);
+            TextView bodyTextName = container.findViewById(R.id.font_card_body_name);
+            bodyTextName.setTypeface(mBodyFont);
+            bodyTextName.setText(mLabel);
             container.findViewById(R.id.font_card_divider).setBackgroundColor(
                     title.getCurrentTextColor());
         }
 
         @Override
         public Builder buildStep(Builder builder) {
-            builder.setHeadlineFontFamily(mHeadlineFont).setBodyFontFamily(mBodyFont);
+            builder.setHeadlineFontFamily(mHeadlineFont)
+                    .setBodyFontFamily(mBodyFont).setFontName(mLabel);
             return super.buildStep(builder);
         }
     }
diff --git a/src/com/android/customization/picker/theme/ThemeFragment.java b/src/com/android/customization/picker/theme/ThemeFragment.java
index 641d3e1..0cad626 100644
--- a/src/com/android/customization/picker/theme/ThemeFragment.java
+++ b/src/com/android/customization/picker/theme/ThemeFragment.java
@@ -400,6 +400,9 @@ public class ThemeFragment extends ToolbarFragment {
                     title.setTypeface(previewInfo.headlineFontFamily);
                     TextView body = card.findViewById(R.id.font_card_body);
                     body.setTypeface(previewInfo.bodyFontFamily);
+                    TextView name = card.findViewById(R.id.font_card_body_name);
+                    name.setTypeface(previewInfo.bodyFontFamily);
+                    name.setText(previewInfo.fontName);
                     card.findViewById(R.id.font_card_divider).setBackgroundColor(accentColor);
                 }
             });
-- 
2.17.1

